### ISSUE TITLE
Slack dm to errbot causes error

### ISSUE THREAD

**Issue by rkrambovitis** [errbotio/errbot#949]
### I am...

* [x] Reporting a bug
* [ ] Suggesting a new feature
* [ ] Requesting help with running my bot
* [ ] Requesting help writing plugins
* [ ] Here about something else

### I am running...

* Errbot version: latest - git
* OS version: Ubuntu 16.04
* Python version: 3.5
* Using a virtual environment: yes (docker)

### Issue description

When issuing command via DM, an exception is thrown.

```
ERROR    errbot.backends.slack     message event handler raised an exception
Traceback (most recent call last):
  File "/err/virtualenv/lib/python3.5/site-packages/errbot/backends/slack.py", line 417, in _dispatch_slack_message
    event_handler(message)
  File "/err/virtualenv/lib/python3.5/site-packages/errbot/backends/slack.py", line 499, in _message_event_handler
    msg.to = SlackPerson(self.sc, self.username_to_userid(self.sc.server.username),
  File "/err/virtualenv/lib/python3.5/site-packages/errbot/backends/slack.py", line 547, in username_to_userid
    raise UserDoesNotExistError("Cannot find user %s" % name)
errbot.backends.base.UserDoesNotExistError: Cannot find user errbot
```

### Steps to reproduce

Use Slack, issue command to errbot using DM.

**Comment by zoni** [comment]
I'm unable to reproduce this. Is this happening for all users in your organization or just one person? Are you able to supply `DEBUG`-level log messages leading up to the exception?

**Comment by rkrambovitis** [comment]
* This affects all users (it was originally reported to me by a user).

The debug log doesn't really add much:
```
16:15:34 DEBUG    errbot.backends.slack     Processing slack event: {'team': 'xxxxxxxxx', 'user': 'Uxxxxxxxx', 'text': '#cds ops', 'channel': 'xxxxxxxxx', 'type': 'message', 'ts': '1485180934.000058'}
16:15:34 DEBUG    errbot.backends.slack     Saw an event: {'channel': 'xxxxxxxxx',
 'team': 'xxxxxxxxx',
 'text': '#cds ops',
 'ts': '1485180934.000058',
 'type': 'message',
 'user': 'Uxxxxxxxx'}
16:15:34 DEBUG    errbot.backends.slack     Escaped IDs event text: #cds ops
16:15:34 ERROR    errbot.backends.slack     message event handler raised an exception
Traceback (most recent call last):
  File "/err/virtualenv/lib/python3.5/site-packages/errbot/backends/slack.py", line 417, in _dispatch_slack_message
    event_handler(message)
  File "/err/virtualenv/lib/python3.5/site-packages/errbot/backends/slack.py", line 499, in _message_event_handler
    msg.to = SlackPerson(self.sc, self.username_to_userid(self.sc.server.username),
  File "/err/virtualenv/lib/python3.5/site-packages/errbot/backends/slack.py", line 547, in username_to_userid
    raise UserDoesNotExistError("Cannot find user %s" % name)
errbot.backends.base.UserDoesNotExistError: Cannot find user errbot
16:15:34 DEBUG    errbot.backends.slack     No event handler available for desktop_notification, ignoring this event
```

Pls note, the "user" not found is itself.
i.e. errbot, not the user issuing the command.

**Comment by zoni** [comment]
Does this issue persist after the bot has been restarted?

**Comment by rkrambovitis** [comment]
yup, the problem remains after any restarts and updates.

**Comment by joharohl** [comment]
I had the same issue. Downgrading slackclient to 1.0.4 solved it for me.

**Comment by davvi** [comment]
Can confirm behavior.

`2017-01-29 22:51:11,190 ERROR    errbot.backends.slack     message event handler raised an exception
Traceback (most recent call last):
  File "/mnt/data/srv/errbot/lib/python3.6/site-packages/errbot/backends/slack.py", line 417, in _dispatch_slack_message
    event_handler(message)
  File "/mnt/data/srv/errbot/lib/python3.6/site-packages/errbot/backends/slack.py", line 499, in _message_event_handler
    msg.to = SlackPerson(self.sc, self.username_to_userid(self.sc.server.username),
  File "/mnt/data/srv/errbot/lib/python3.6/site-packages/errbot/backends/slack.py", line 547, in username_to_userid
    raise UserDoesNotExistError("Cannot find user %s" % name)
errbot.backends.base.UserDoesNotExistError: Cannot find user bender
2017-01-29 22:51:15,905 DEBUG    errbot.backends.slack     No event handler available for desktop_notification, ignoring this event
2017-01-29 22:51:31,082 DEBUG    errbot.backends.slack     No event handler available for user_typing, ignoring this event
2017-01-29 22:51:34,169 DEBUG    errbot.backends.slack     No event handler available for user_typing, ignoring this event
2017-01-29 22:51:34,828 DEBUG    errbot.backends.slack     Processing slack event: {'type': 'message', 'channel': 'D3X9P6SSF', 'user': 'U20PH289H', 'text': 'plugin activate Bender', 'ts': '1485690696.000030', 'team': 'T20R5TEJK'}
2017-01-29 22:51:34,829 DEBUG    errbot.backends.slack     Saw an event: {'channel': 'D3X9P6SSF',
 'team': 'T20R5TEJK',
 'text': 'plugin activate Bender',
 'ts': '1485690696.000030',
 'type': 'message',
 'user': 'U20PH289H'}
2017-01-29 22:51:34,829 DEBUG    errbot.backends.slack     Escaped IDs event text: plugin activate Bender
2017-01-29 22:51:34,829 ERROR    errbot.backends.slack     message event handler raised an exception
Traceback (most recent call last):
  File "/mnt/data/srv/errbot/lib/python3.6/site-packages/errbot/backends/slack.py", line 417, in _dispatch_slack_message
    event_handler(message)
  File "/mnt/data/srv/errbot/lib/python3.6/site-packages/errbot/backends/slack.py", line 499, in _message_event_handler
    msg.to = SlackPerson(self.sc, self.username_to_userid(self.sc.server.username),
  File "/mnt/data/srv/errbot/lib/python3.6/site-packages/errbot/backends/slack.py", line 547, in username_to_userid
    raise UserDoesNotExistError("Cannot find user %s" % name)
errbot.backends.base.UserDoesNotExistError: Cannot find user bender`

Configured with  a slack bot integration ..I have named my bot bender in slack.

**Comment by zoni** [comment]
I have yet to confirm this (so lets not collectively jump to conclusions just yet) but I suspect slackclient made yet another breaking API change (they did so with the [1.0.2 → 1.0.3](https://github.com/errbotio/errbot/pull/936) release as well) recently, specifically in https://github.com/slackapi/python-slackclient/pull/152 or https://github.com/slackapi/python-slackclient/pull/156.

If people affected by this bug could verify there is no issue with slackclient version 1.0.3 (`pip install slackclient==1.0.3`) that would be invaluable feedback as it would demonstrate it is indeed caused by different behaviour of this library.

**Comment by rkrambovitis** [comment]
After a little testing.
 - 1.0.2 works fine
 - 1.0.3 does not work correctly. The command permissions fail.
```
08:45:48 ERROR    errbot.backends.slack     Cannot find user with ID Uxxxxxxxx
08:45:48 INFO     errbot.plugins.ACLS       Matching ACL {'allowusers': ('@user1', '@user2', '@user3', '@user4', '@user5')} against username @<Uxxxxxxxx> for command foo:bar
08:45:48 ERROR    errbot.backends.slack     Cannot find user with ID Uxxxxxxxx
08:45:48 INFO     errbot.core               Command None blocked or deferred.

```

 - 1.0.4 appears to be working fine.
 - 1.0.5 throws the error I originally posted.

**Comment by br0ziliy** [comment]
I can confirm that this issue absent when using slackclient 1.0.4.
Specific change that introduced this issue is this one (as correctly pointed out by @zoni): https://github.com/slackapi/python-slackclient/pull/156

**Comment by rcoup** [comment]
Dupe of https://github.com/errbotio/errbot/issues/956?

**Comment by zoni** [comment]
> Dupe of #956?

I'd say the other way around. This issue was filed earlier. ;)

Changing jobs to [StartMail](https://www.startmail.com/) prevented me from getting around to this faster, but I just had some spare time this evening to tackle the issue. If people could confirm the fix in #961 works for them I can commit that to master and we can do another release to get this out to everyone.

**Comment by rkrambovitis** [comment]
Yup, works fine for me :)

Many thanks :)

**Comment by zoni** [comment]
Great! We'll work on getting a new release out soon with that included.

**Comment by gbin** [comment]
fixed in 4.3.7
